test/h2spec/client.go

```diff
@@ -0,0 +1,155 @@
+package h2spec
+
+// This is a slightly modified version of https://github.com/summerwind/h2spec/blob/70ac2294010887f48b18e2d64f5cccd48421fad1/cmd/h2spec/h2spec.go
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/summerwind/h2spec"
+	h2specconfig "github.com/summerwind/h2spec/config"
+)
+
+var (
+	version string = "2.6.0"
+	commit  string = "70ac2294010887f48b18e2d64f5cccd48421fad1"
+)
+
+func NewClientCommand() *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:   "h2spec [args...]",
+		Short: "Conformance testing tool for HTTP/2 implementation",
+		Long:  "Conformance testing tool for HTTP/2 implementation.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return run(cmd, args)
+		},
+	}
+
+	cmd.SilenceUsage = true
+	cmd.SilenceErrors = true
+
+	flags := cmd.Flags()
+	flags.StringP("host", "h", "127.0.0.1", "Target host")
+	flags.IntP("port", "p", 0, "Target port")
+	flags.StringP("path", "P", "/", "Target path")
+	flags.IntP("timeout", "o", 2, "Time seconds to test timeout")
+	flags.Int("max-header-length", 4000, "Maximum length of HTTP header")
+	flags.StringP("junit-report", "j", "", "Path for JUnit test report")
+	flags.BoolP("strict", "S", false, "Run all test cases including strict test cases")
+	flags.Bool("dryrun", false, "Display only the title of test cases")
+	flags.BoolP("tls", "t", false, "Connect over TLS")
+	flags.StringP("ciphers", "c", "", "List of colon-separated TLS cipher names")
+	flags.BoolP("insecure", "k", false, "Don't verify server's certificate")
+	flags.BoolP("verbose", "v", false, "Output verbose log")
+	flags.Bool("version", false, "Display version information and exit")
+	flags.Bool("help", false, "Display this help and exit")
```

Comment:
`help` and `version` won't ever be used, will they?

Reply:
Also - where is the `yaml` change for this as an ingress-operator  subcommand?

Reply:
> `help` and `version` won't ever be used, will they?

Depends if we ever ask a customer to run this. If so, they will be useful. We can verify which version, and the help explains what the options are.

Reply:
> Also - where is the `yaml` change for this as an ingress-operator subcommand?

There's no associated YAML for this. Perhaps I'm missing the question here?

Reply:
It's been merged now, but I think I was looking for deployment yaml to be updated.  I guess it is not needed.

Reply:
I believe the h2spec client code is only going to be consumed by tests in origin. So no changes are needed to the ingress-operator or router deployment yaml.